mflow.go

```diff
@@ -0,0 +1,80 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"sort"
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+	prom "github.com/prometheus/client_golang/prometheus"
+)
+
+type labelNameValue struct {
+	Name  string `json:"name"`
+	Value string `json:"value"`
+}
+
+type metricUpdateEvent struct {
+	Name                string           `json:"name"`
+	LabelNameValuePairs []labelNameValue `json:"labelNameValuePairs"`
+	Value               float64          `json:"value"`
+}
+
+// sort.Interface
+type byLabelName []labelNameValue
+
+func (a byLabelName) Len() int           { return len(a) }
+func (a byLabelName) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
+func (a byLabelName) Less(i, j int) bool { return a[i].Name < a[j].Name }
+
+func sortedLabelNames(labelNameValuePairs []labelNameValue) []string {
```

Comment:
In Go 1.8 you can use sort.Slice here, though You've already typed in the code so it doesn't much matter ;)